feegrant/index.md

```diff
@@ -0,0 +1,254 @@
+
+<!-- TODO: update tip to use `unsafe-reset-all` with better user experience -->
+
+::: tip
+If you have used `simd` before, you might already have a `.simapp` directory in your home directory. To keep the previous data, either save the directory to another location or use the `--home` flag and specify a different directory for each command in the following instructions. If you don't want to keep the previous data, remove the previous directory (`rm -rf ~/.simapp`).
+:::
+
+Run the following commands to configure the `simd` binary.
+
+Set the chain ID:
+
+```bash
+simd config chain-id demo
+```
+
+Set the [keyring backend](https://docs.cosmos.network/v0.42/run-node/keyring.html#the-test-backend):
+
+```bash
+simd config keyring-backend test
+```
+
+## Key Setup
+
+You'll have to create a few test keys for your users.
+
+Add a key for Alice, the validator:
+
+```bash
+simd keys add alice
+```
+
+Add a key for Bob, the grantee:
+
+```bash
+simd keys add bob
+```
+
+And lastly, add a key for Carol, the granter:
+
+```bash
+simd keys add carol
+```
+
+If you'd like to see an overview of your keys, use:
+
+```bash
+simd keys list
+```
+
+::: tip
+To avoid having to copy and paste the user addresses now is a good time to export the user keys to variables that you can access and use for this tutorial. 
+:::
+
+```bash
+export ALICE_KEY=$(simd keys show alice -a)
+export BOB_KEY=$(simd keys show bob -a)
+export CAROL_KEY=$(simd keys show carol -a)
+```
+
+## Chain Setup
+
+The following commands set up a chain using the simulation application (`simapp`).
+
+Initialize the node:
+
+```bash
+simd init test --chain-id demo
+```
+
+Alice is your validator. Add Alice and an initial balance to the genesis file:
+
+```bash
+simd add-genesis-account alice 5000000000stake --keyring-backend test
+```
+
+Add Bob and an initial balance to the genesis file:
+
+```bash
+simd add-genesis-account bob 2000kudos --keyring-backend test
+```
+
+Note that Bob has only `kudos` tokens and won't be able to pay for any fees that might be needed.
+
+Add Carol and an initial `stake` balance to the genesis file:
+
+```bash
+simd add-genesis-account carol 1000000stake --keyring-backend test
+```
+
+Generate a transaction to add Alice to the initial validator set:
+
+```bash
+simd gentx alice 1000000stake --chain-id demo
+```
+
+Add the validator transaction to the genesis file:
+
+```bash
+simd collect-gentxs
+```
+
+## Start Chain
+
+You are now ready to start a single node network on your local machine.
+
+Start the chain:
+
+```bash
+simd start
+```
+
+## Grant Allowance
+
+Before Bob can send `kudos` to Carol, you must set up an allowance for Bob so that Carol pays for any gas fees the transaction might incur. Because Alice is the validator, it does not make sense to make her the granter because a validator receives all the fees that the granter paid for. If Alice were the granter, there would not be a difference in her balance after a transaction was completed.
```

Comment:
Alice does not directly receive the fees from the transaction. For example, when you run the balance query a second time after executing the send transaction, Alice's balance is unchanged.

Validators must claim their rewards through a separate transaction. Also, block rewards are distributed proportionally to all validators relative to their voting power. Yes, Alice is the only validator in this case but this statement might cause some confusion (i.e. "a validator receives all the fees").

I think this statement is a bit confusing and the example could in fact be done with only two accounts, one account being both the validator and granter (or grantee). I think three accounts works too and the separation might help beginners but I would remove or revise the second part of this paragraph as it might cause confusion.

```suggestion
Before Bob can send `kudos` to Carol, you must set up an allowance for Bob so that Carol pays for any gas fees the transaction might incur.
```

Reply:
AH! That's a really good point @ryanchristo , thanks for highlighting! This does make me wonder whether or not to just stick to two accounts and reduce the complexity. I think there's no reason to stick to three accounts really. Wdyt?

Reply:
Up to you! I probably would have done two but three works too.